utils.ts

```diff
@@ -58,15 +58,19 @@ export function serialize(val: string): u32 {
 export function deserialize(encoded_allocation: u32): string {
   let offset = u32_high_bits(encoded_allocation);
   let length = u32_low_bits(encoded_allocation);
-  let res: string = allocateUnsafe(length);
+  let res    = allocateUnsafe(length);
 
   // TODO: figure out how to do this in a single copy. Need to change boundaries on characters
-  for (let i: u16 = 0; i < length; i++) {
+  /* for (let i = 0; i < length; ++i) {
     memory.copy(
```

Comment:
Deserialize needs to convert a byte array that is a utf-8 encoded string to a assemblyscript encoded string. What would be the best way to do this in your opinion?

Reply:
You could do as I proposed in [this line](https://github.com/holochain/hdk-assemblyscript/pull/6/files#diff-5275ea9aafad79f2f6b3d7a4abe777a8R72). This would be works with ASCII and Latin1 but not for other languages. If you need true UTF-8 support you should use much complicated approach and implement utf8 -> utf16 method. AS implemented only utf16 -> utf8 for now